File: brain/bbafd75a-b6c0-4533-b9f1-20c720729b4e/scratch/check_tags.py
import re

def count_tags(filename):
    with open(filename, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Simple regex for opening and closing tags
    # This is naive but might help find obvious mismatches
    opening_tags = re.findall(r'<([a-zA-Z0-9]+)(?:\s+[^>]*[^/])?>', content)
    closing_tags = re.findall(r'</([a-zA-Z0-9]+)>', content)
    
    counts = {}
    for tag in opening_tags:
        counts[tag] = counts.get(tag, 0) + 1
    for tag in closing_tags:
        counts[tag] = counts.get(tag, 0) - 1
    
    for tag, count in counts.items():
        if count != 0:
            print(f"Tag <{tag}> mismatch: {count}")

count_tags('d:/systems/bantudefdig/frontend/src/features/developer_admin/TenantsPage.tsx')

File: brain/bbafd75a-b6c0-4533-b9f1-20c720729b4e/scratch/check_tags.js
const fs = require('fs');

const content = fs.readFileSync('d:/systems/bantudefdig/frontend/src/features/developer_admin/TenantsPage.tsx', 'utf8');

const openingTags = content.match(/<([a-zA-Z0-9]+)(?:\s+[^>]*[^/])?>/g) || [];
const closingTags = content.match(/<\/([a-zA-Z0-9]+)>/g) || [];

const counts = {};

openingTags.forEach(tag => {
    const tagName = tag.match(/<([a-zA-Z0-9]+)/)[1];
    counts[tagName] = (counts[tagName] || 0) + 1;
});

closingTags.forEach(tag => {
    const tagName = tag.match(/<\/([a-zA-Z0-9]+)/)[1];
    counts[tagName] = (counts[tagName] || 0) - 1;
});

for (const [tag, count] of Object.entries(counts)) {
    if (count !== 0) {
        console.log(`Tag <${tag}> mismatch: ${count}`);
    }
}
